### ISSUE TITLE
class extends interface not caught

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1435]
# repro
* using ea14933
* https://psalm.dev/r/0af0873933
* https://3v4l.org/KCCEF

```php
<?php
interface Foo {}

class Bar extends Foo {}
```

## expected
something to the effect of "class cannot extend from interface" as per runtime fatal error.

## actual
> Psalm output (using commit ea14933): 
> 
> No issues!

**Comment by weirdan** [comment]
Also the other way around:
```php
<?php
class Foo {}
class Bar implements Foo {}
```
https://psalm.dev/r/29a62a4a3d
